Configure log format

janisz opened the pull request.

dankraw commented on config/config.go:
```diff
@@ -32,7 +36,13 @@ func New() (*Config, error) {
-	config.setLogLevel()
+
+	if err != nil {
+		return config, err
```
I think we should return `nil, err` here as config is already defined as `Config{}`

janisz replied:
Fixed

dankraw commented:
:+1:

janisz commented:
We have Marathon password leak in logs. I fixed it and also make logs more consistent every log (if possible) has `"Id"` field which contains Task ID or (if not exist in current context) Application ID. Also added some more debug logs.

dankraw commented on config/config.go:
```diff
@@ -68,8 +82,11 @@ func (config *Config) parseFlags() {
 
+	// Log
+	flag.StringVar(&config.Log.Level, "log-level", "info", "Log level: panic, fatal, error, warn, info, or debug")
+	flag.StringVar(&config.Log.Format, "log-format", "text", "Log format: JSON, text")
```
in code we check against "TEXT" (uppercase) - we should stick to the rule here.
